imagedephi/redact/dicom.py

```diff
@@ -84,13 +86,25 @@ def __init__(self, image_path: Path, rules: DicomRules) -> None:
         self.metadata_redaction_steps = {}
         self.no_match_tags = []
         self.uid_map = {}
-        print(self.dicom_data)
 
         for element, _ in DicomRedactionPlan._iter_dicom_elements(self.dicom_data):
+            custom_metadata_key = "CustomMetadataItem"
             keyword = keyword_for_tag(element.tag)
-            keyword_in_rules = keyword in rules.metadata
-            if not keyword_in_rules:
-                self.no_match_tags.append(element.tag)
+            # Check keyword and (gggg,eeee) representation
+            tag_in_rules = keyword in rules.metadata or str(element.tag) in rules.metadata
+            if not tag_in_rules:
+                if element.tag.group % 2 == 1:
+                    if rules.delete_custom_metadata:
+                        # If the group is odd, it is custom metadata. Use the custom metadata action
+                        self.metadata_redaction_steps[element.tag] = DeleteRule(
+                            key_name=custom_metadata_key, action="delete"
+                        )
+                    else:
+                        self.metadata_redaction_steps[element.tag] = KeepRule(
```

Comment:
I think this ultimately should have three states, delete custom metadata, keep custom metadata, or do not have auto rules on custom metadata.  This could be a future PR, though.  Maybe it is a pair of booleans: delete_custom_metadata and keep_custom_metadata which would then make no sense if both true.